lib/travis/build/script/r.rb

```diff
@@ -217,11 +249,15 @@ def r_github_install(packages)
           ].join(' ')
           sh.cmd "Rscript -e '#{install_script}'"
         end
-        
+
         def r_binary_install(packages)
           return if packages.empty?
-          case config[:os]
-          when 'linux'
+          if config[:os] == 'linux'
+            if config[:sudo] == false
```

Comment:
This is not an entirely accurate way to check if this build can use `sudo` or not. This is mostly for the historical reason; older repositories defaults to `sudo: required`, while new ones do to `sudo: false`. In other words, the build can lack `sudo` without `sudo: false`.

I am not sure if precise detection is possible from here, unfortunately.

I also believe that the absence of `sudo` should result in `config[:sudo]` being `nil`, so this clause would be skipped.

Perhaps you can approximate by `unless config[:sudo]` (with some false positives related to the default behavior depending on the age of the repository).